test/changelog_web/controllers/search_controller_test.exs

```diff
@@ -19,12 +25,20 @@ defmodule ChangelogWeb.SearchControllerTest do
     test "getting the search with one result" do
       item1 = insert(:published_news_item, headline: "Phoenix", story: "oh my")
       item2 = insert(:published_news_item, headline: "Rails", story: "my oh")
-      results = %{"hits" => [item_to_hit(item1)], "nbHits" => 1}
+      results = %{
+        "hits" => [item_to_hit(item1)],
+        "found" => 1,
+        "out_of" => 100,
+        "page" => 1,
+        "request_params" => %{
+          "per_page" => 10
+        }
+      }
 
-      with_mock(Algolia, search: fn _, _, _ -> {:ok, results} end) do
+      with_mock(Typesense.Client, search: fn _, _ -> {:ok, results} end) do
         conn = get(build_conn(), Routes.search_path(build_conn(), :search, q: "phoenix"))
 
-        assert called(Algolia.search(:_, "phoenix", :_))
+        assert called(Typesense.Client.search(:_, %{q: "phoenix"}))
```

Comment:
Change `%{q: "phoenix"}` to `:meck.is(fn opts -> opts[:q] == "phoenix" end)` and this will pass

Reply:
Thanks @jerodsanto! I've updated it, and all tests pass now.